platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/StsRealm.java

```diff
@@ -13,14 +13,665 @@
+import org.codice.ddf.configuration.PropertyResolver;
+import org.codice.ddf.security.handler.api.BaseAuthenticationToken;
+import org.codice.ddf.security.handler.api.SAMLAuthenticationToken;
+import org.codice.ddf.security.policy.context.ContextPolicy;
+import org.codice.ddf.security.policy.context.ContextPolicyManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.DOMImplementation;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSSerializer;
+
+public class StsRealm extends AuthenticatingRealm implements STSClientConfiguration {
+  private static final Logger LOGGER = (LoggerFactory.getLogger(StsRealm.class));
+
+  private static final String NAME = StsRealm.class.getSimpleName();
+
+  private static final String ADDRESSING_NAMESPACE = "http://www.w3.org/2005/08/addressing";
+
+  private static final Splitter SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+
+  protected Bus bus;
+
+  PropertyResolver address = null;
+
+  String endpointName = null;
+
+  String serviceName = null;
+
+  String username = null;
+
+  String password = null;
+
+  String signatureUsername = null;
+
+  String signatureProperties = null;
+
+  String encryptionUsername = null;
+
+  String encryptionProperties = null;
+
+  String tokenUsername = null;
+
+  String tokenProperties = null;
+
+  List<String> claims = new ArrayList<>();
+
+  private ContextPolicyManager contextPolicyManager;
+
+  private String assertionType = null;
+
+  private String keyType = null;
+
+  private String keySize = null;
+
+  private Boolean useKey = null;
+
+  private Cache<Element, SecurityToken> cache =
+      CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.MINUTES).build();
+
+  public StsRealm() {
+    this.bus = getBus();
+    setCredentialsMatcher(new STSCredentialsMatcher());
+  }
+
+  public ContextPolicyManager getContextPolicyManager() {
+    return contextPolicyManager;
+  }
+
+  public void setContextPolicyManager(ContextPolicyManager contextPolicyManager) {
+    this.contextPolicyManager = contextPolicyManager;
+  }
+
+  /** Determine if the supplied token is supported by this realm. */
+  @Override
+  public boolean supports(AuthenticationToken token) {
+    boolean supported = token != null && token.getCredentials() != null;
+
+    if (supported) {
+      LOGGER.debug("Token {} is supported by {}.", token.getClass(), StsRealm.class.getName());
+    } else if (token != null) {
+      LOGGER.debug("Token {} is not supported by {}.", token.getClass(), StsRealm.class.getName());
+    } else {
+      LOGGER.debug("The supplied authentication token is null. Sending back not supported.");
+    }
+
+    return supported;
+  }
+
+  /** Perform authentication based on the supplied token. */
+  @Override
+  protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token) {
```

Comment:
❓ When looking in intellij, it appears this method is only used in test classes. Is that expected?

Reply:
It's called by Shiro